core-war/src/main/java/org/silverpeas/web/pdc/control/PdcSearchSessionController.java

```diff
@@ -547,23 +572,32 @@ private void processFacetsFormField(Map<String, Facet> fieldFacetsMap, GlobalSil
           facet = fieldFacetsMap.get(facetId);
         }
         if (facet != null) {
+          FacetEntryVO entry = null;
           String fieldValueKey = fieldsForFacets.get(facetId);
-          String fieldValueLabel = fieldValueKey;
-          try {
-            fieldValueLabel = getFieldValue(formName, fieldName, fieldValueKey);
-          } catch (Exception e) {
-            SilverLogger.getLogger(this).error(e.getMessage(), e);
+          if (facet instanceof FacetOnDates) {
+            entry = ((FacetOnDates) facet).addEntry(fieldValueKey);
+          } else {
+            String fieldValueLabel = fieldValueKey;
+            try {
+              fieldValueLabel = getFieldValue(formName, fieldName, fieldValueKey);
+            } catch (Exception e) {
+              SilverLogger.getLogger(this).error(e.getMessage(), e);
+            }
+            entry = new FacetEntryVO(fieldValueLabel, fieldValueKey);
+            facet.addEntry(entry);
           }
-          FacetEntryVO entry = new FacetEntryVO(fieldValueLabel, fieldValueKey);
           // check if this entry have been selected
           if (getSelectedFacetEntries() != null) {
             String selectedEntry =
                 getSelectedFacetEntries().getFormFieldSelectedFacetEntry(facet.getId());
-            if (StringUtil.isDefined(selectedEntry) && selectedEntry.equals(fieldValueKey)) {
-              entry.setSelected(true);
+            boolean selected = false;
+            if (facet instanceof FacetOnDates) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS134)